### PR TITLE
enhancement(switcher): implementing dark-light switcher for remote

### PR DESCRIPTION
**Pull request by grantlouisherman**
hey @peterpeterparker let me know what you think.

### CODE REVIEW

**Comment by peterpeterparker** [comment]
@grantlouisherman really cool!

before merging (the switcher itself looks neat 👍 ) I have just have one request, if you allow me. 

instead of defining a new `.dark` style, could you use the already defined default dark mode (defined with [@media (prefers-color-scheme: dark)](https://github.com/deckgo/deckdeckgo/blob/bda129a6844c40bfab457686d6db937bab6bea43/remote/src/global/variables.scss#L66)) ?

**Comment by grantlouisherman** [comment]
Hey @peterpeterparker so it doesn't seem like I can set media types with js, but I can add something that if that media exist to set the toggle to dark. Unless you know of a way to set a media with js? I just couldn't find a way. It seemed like all the material I read was more for detecting if that media preference existed as opposed to setting one

**Comment by peterpeterparker** [comment]
@grantlouisherman I'll have a look tomorrow morning and will keep you posted

**Comment by grantlouisherman** [comment]
This is what I found in terms of interaction with media queries: https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList

It seems like you can add listeners or remove listeners, but you can specifically add it. One example I saw was to create a separate style sheet with the preferred media behavior and inject that into the DOM , but I think that is over engineering the problem.

Let me know what you find.

**Comment by peterpeterparker** [comment]
@grantlouisherman I think the media query listener is probably the way. When the app load, we should check the current state if dark or light and then if dark, set the `.dark` as you do in your PR. Of course only if the user didn't already selected a style using the toggler.

Then, to finish the transformation, we should move the current css from `@media (prefers-color-scheme: dark)` to `.dark`

So basically we only use the `prefers-color-scheme` to test the platform preferences but to apply the css we use `.dark`.

Does that make sense you think?

File: remote/src/app/app-root.scss [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    --ion-item-background-activated: #313131;
+    }
+
+@media (prefers-color-scheme: dark) {
+    :root {
+        --ion-background-color: black;
+        --ion-text-color: #ffffff;
+        --ion-item-background: #1A1B1E;
+        --ion-item-background-activated: #313131;
+        }
```
Comment (grantlouisherman) [inline remote/src/app/app-root.scss]: @peterpeterparker 👀  here

File: remote/src/app/modals/app-remote-dark-mode-switch/app-remote-dark-mode-switch.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    private resolveDarkModePreference(): Boolean | Promise<any> {
+        let getDarkModePreferenceFromIdb: Promise<any> = this.idbKeyValPromiseHandler(get('darkModePreferences'));
+        let getDarkModePreferenceFromMedia = window.matchMedia('(prefers-color-scheme: dark)');
+        if( getDarkModePreferenceFromIdb instanceof Error ){
+            return false;
+        }
+        let isDarkModePreferenceDetermined: Boolean | Promise<any> = 
+        getDarkModePreferenceFromIdb || 
+        getDarkModePreferenceFromMedia.matches;
```
Comment (grantlouisherman) [inline remote/src/app/modals/app-remote-dark-mode-switch/app-remote-dark-mode-switch.tsx]: @peterpeterparker 👀 so we could do something like this where we check for the dark mode preference when the application is loaded. So here we are checking the media query and the key val stuff. What do you think? If you wanted to added a full listener I could add a @watch event for `onchange`? Im not super well versed in stencil, but I think that could work?

Comment (peterpeterparker) [reply remote/src/app/modals/app-remote-dark-mode-switch/app-remote-dark-mode-switch.tsx]: Yes exactly we could do something like this. The remote UX will be refactored, so based on that following implementation we will be able to fine tune the solution.

Just note, currently, `this.darkModePreference = await this.resolveDarkModePreference();` resolve `undefined` in my case which doesn't seems correct. Maybe a promise not resolving?

**Comment by peterpeterparker** [comment]
Awesome @grantlouisherman 👍  Thx a lot for the amazing work 🙏

I'll continue this in the UX refactoring #228 but damn it's cool!
